undofx/src/main/java/org/fxmisc/undo/impl/UndoManagerImpl.java

```diff
@@ -209,8 +229,10 @@ private void addChange(C change) {
     @SuppressWarnings("unchecked")
     private C[] merge(C c1, C c2) {
         Optional<C> merged = merge.apply(c1, c2);
-        if(merged.isPresent()) {
-            return (C[]) new Object[] { merged.get() };
+        if(merged.isPresent() && isNotIdentity.test(merged.get())) {
+            return isNotIdentity.test(merged.get())
```

Comment:
WOW!.... Can't believe I forgot to remove the first test and only have the second... Sheez...